pageserver/src/layered_repository/timeline.rs

```diff
@@ -835,92 +928,157 @@ impl Timeline {
         Ok(())
     }
 
-    /// (Re-)calculate the logical size of the database at the latest LSN.
+    /// Retrieve current logical size of the timeline.
     ///
-    /// This can be a slow operation.
-    pub fn init_logical_size(&self) -> Result<()> {
-        if self.logical_size_initialized.load(AtomicOrdering::Acquire) {
-            return Ok(());
+    /// The size could be lagging behind the actual number, in case
+    /// the initial size calculation has not been run (gets triggered on the first size access).
```

Comment:
So if the initial size hasn't been calculated yet, this returns a bogus value?

Also, if the current "partial size" is negative, current_size() will return error, and this function will also return an Error.

I'd suggest always returning 0 or None if the initial calculation hasn't finished yet.

Reply:
> Not sure how realistic this scenario is, but same doubts I have about the whole init in the background approach (with the new layer sizes, seems that init happens in reasonable time for it to be moved away from the background task to the main code).

FYI https://github.com/neondatabase/neon/pull/2265#discussion_r946129370

I tried to check this for bigger project size on staging, but custom limit setting is broken https://github.com/neondatabase/cloud/issues/2198

For smaller databases IMO this is definitely not a problem anymore